### ISSUE TITLE
Filebeat AWS Module - S3 input support for AWS WAF logs

### ISSUE THREAD

**Issue by octavialarentisgsa** [elastic/beats#28121]
**Describe the enhancement:**
An enhancement to current the Filebeat AWS module to allow parsing of AWS WAF logs directly to ECS format is requested.

**Describe a specific use case for the enhancement or feature:**
Currently the Filebeat AWS module allows certain AWS logs to be pulled directly from AWS S3 buckets (CloudTrail, CloudWatch, ELB, EC2 etc.). This module does not offer support for WAF logs, which requires custom log format parsing to migrate the raw log format into the ECS format.

**Comment by legoguy1000** [comment]
@octavialarentisgsa  can you take a look at elastic/integrations#1886.  Since you use the WAF logs, I'd like to get your input.

**Comment by kaykhan** [comment]
Its a little bit confusing from the current documentation how to actually integrate aws waf logs into elasticsearch. 

I still don't see waf mentioned under the filebeat module https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-aws.html - so is it even possible? although it is mentioned in the elastic agents https://docs.elastic.co/en/integrations/aws/waf (we don't use elastic agents)

**Comment by legoguy1000** [comment]
Correct, there is no filebeat module for it. Only an agent integration.

**Comment by kaykhan** [comment]
So when the documentation says `Prefer to use Beats for this use case? See [Filebeat modules](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-modules-overview.html) for logs or [Metricbeat modules](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-modules.html) for metrics.`

What is it suggesting?

**Comment by legoguy1000** [comment]
That's a generic message for agent vs filebeat/metricbeat in general not for each specific integration.

**Comment by legoguy1000** [comment]
You could easily take the components from the waf integration and apply it to filebeat.

**Comment by kaykhan** [comment]
Okay that sucks because it suggests new features are being developed fro elastic agents and not for filebeat/metricbeat now. The reasons we haven't switched is because elastic agents are not as configurable as filebeat and metricbeat and elastic agents is alpha with very little support with Elastic Cloud On Kubenretes :/

**Comment by kaykhan** [comment]
> .

Not sure what you mean by `take components from waf integration` Can you point me to what you mean?

**Comment by kaykhan** [comment]
> You could easily take the components from the waf integration and apply it to filebeat.

Not sure what you mean by `take components from waf integration` Can you point me to what you mean?

**Comment by legoguy1000** [comment]
So first, the agent is just a mgmt process running filebeat, metricbeat, packetbeat.... Under the hood. So all the same inputs and capabilities in filebeat are in the agent. The integrations are essentially prepacked components utilizing a beat input(s) with a datastream, pipelines, mappings..... So there are definitely new features being added to beats, not necessarily new modules since agent integrations are the next evolution of beats modules.

The AWS WAF integration is just the aws-s3 input from filebeat, an index field mapping, and an ingest pipeline. U can load the pipeline manually, creat a template with those fields, creat a filebeat S3 input that points to that index template/pipeline and done.

**Comment by kaykhan** [comment]
> So first, the agent is just a mgmt process running filebeat, metricbeat, packetbeat.... Under the hood. So all the same inputs and capabilities in filebeat are in the agent. The integrations are essentially prepacked components utilizing a beat input(s) with a datastream, pipelines, mappings..... So there are definitely new features being added to beats, not necessarily new modules since agent integrations are the next evolution of beats modules.
> 

Okay that makes sense

> The AWS WAF integration is just the aws-s3 input from filebeat, an index field mapping, and an ingest pipeline. U can load the pipeline manually, creat a template with those fields, creat a filebeat S3 input that points to that index template/pipeline and done.

If this has been done for elastic integration, is there somewhere i can see the index field mapping/template and ingest pipeline so i can give this a go? ( i think this template/ingest pipeline is what you meant by components)

**Comment by legoguy1000** [comment]
Yes, everything u need is here https://github.com/elastic/integrations/tree/main/packages/aws/data_stream/waf

**Comment by kaykhan** [comment]
Okay i looked into this a little bit but unfortunately i think its out of my depth, i hope there is some resources that can bring these new features to existing filebeat.

**Comment by legoguy1000** [comment]
I cant speak for Elastic, but based on what I've seen and discussions I've had, I don't know that there will be much effort in adding new modules/fileset to filebeat as the agent and integrations is the long term future.
